src/cache-restore.ts

```diff
@@ -29,7 +29,9 @@ export const restoreCache = async (
     );
   }
 
-  const primaryKey = `setup-go-${platform}-go-${versionSpec}-${fileHash}`;
+  const linuxVersion =
+    process.env.RUNNER_OS === 'Linux' ? `${process.env.ImageOS}-` : '-';
+  const primaryKey = `setup-go-${platform}-${linuxVersion}go-${versionSpec}-${fileHash}`;
```

Comment:
LGTM! I do have a question about this line though. It's a small remark, but if the condition in `linuxVersion` is `false`, would there not be a double hyphen (--) inside the key? Again, just a small observation, but worth asking I suppose

Reply:
fixed